### PR TITLE
Login and Registration functional with light refactor.

### PR DESCRIPTION
**Pull request by calebkrauter**
Worked on much of this during team meetings.
- Add login endpoint and request.
- Refactor code.
- Update sql query and getting data from query.

### CODE REVIEW

**Review by jasgcode — APPROVED** [review]

**Review by jimothy-dev — APPROVED** [review]
Reviewed changes. Looks good.

File: server.js [hunk 1]
```diff
@@ -1,12 +1,28 @@
 const express = require('express');
 const cors = require('cors');
-const { insertUser } = require('./src/api/controllers/keyClashQs');
+const { insertUser, getUser } = require('./src/api/controller/queries');
 const app = express();
 const port = process.env.PORT || 5001;
-
+console.log("App started.")
```
Comment (coltenkrauter) [inline server.js:6]: This could be removed or changed to `console.debug`.

File: server.js [hunk 2]
```diff
@@ -1,12 +1,28 @@
...
 app.use(cors());
 app.use(express.json());
 
+app.post('/api/login', async (req, res) => {
+
+  try {
+    const { email, password_hash } = req.body;
```
Comment (coltenkrauter) [inline server.js:13]: Nit: I personally prefer camelCase in JSON, so `passwordHash`.

File: server.js [hunk 3]
```diff
@@ -1,12 +1,28 @@
...
+
+    const result = await getUser(email, password_hash);
+
+    res.status(200).json({ message: 'Login successful', user: result });
```
Comment (coltenkrauter) [inline server.js:17]:
```suggestion
    const user = await getUser(email, password_hash);

    res.status(200).json({ message: 'Login successful', user });
```

File: server.js [hunk 4]
```diff
@@ -1,12 +1,28 @@
...
+
+  } catch (err) {
+    console.log("ERROR")
```
Comment (coltenkrauter) [inline server.js:20]:
```suggestion
console.error(`Login failed with error [${err}]`)
```

File: server.js [hunk 5]
```diff
@@ -1,12 +1,28 @@
...
+    throw err + " can't login."
```
Comment (coltenkrauter) [inline server.js:21]:
```suggestion
throw `Login failed with error [${err}]`
```

File: src/api/controller/queries.js [hunk 6]
```diff
@@ -17,6 +17,27 @@ async function getLeaderboard() {
   }
 }
 
+async function getUser(email, password_hash) {
+  console.log(password_hash);
+  try {
+    const sql = `
+    SELECT email, password_hash FROM userinfo WHERE email = ? AND password_hash = ?`
+    const [rows] = await pool.query(sql, [email, password_hash]);
+    console.log("USER: " + rows);
```
Comment (coltenkrauter) [inline src/api/controller/queries.js:26]:
```suggestion
    console.log("User data from database", rows);
```

**Review by coltenkrauter — COMMENTED** [review]
A few light comments

**Pull request merged**